cmd/packaging/packaging.go

```diff
@@ -162,15 +157,166 @@ func runDockerPackaging(path string, packagingFormat string, command []string) {
+		}
+	}
+	projectName := pubspec.GetPubSpec().Name
+	createPackagingFormatDirectory(t.packagingFormatName, ignore...)
+	dir := packagingFormatPath(t.packagingFormatName)
+	templateData := getTemplateData(projectName, "")
+	templateData["icon"] = executeStringTemplate(t.linuxDesktopFileIconPath, templateData)
+	templateData["exec"] = executeStringTemplate(t.linuxDesktopFileExecutablePath, templateData)
+	for sourceFile, destinationFile := range t.templateFiles {
+		destinationFile = executeStringTemplate(filepath.Join(dir, destinationFile), templateData)
+		err := os.MkdirAll(filepath.Dir(destinationFile), 0775)
+		if err != nil {
+			log.Errorf("Failed to create directory %s: %v", filepath.Dir(destinationFile), err)
+			os.Exit(1)
+		}
+		fileutils.ExecuteTemplateFromAssetsBox(fmt.Sprintf("packaging/%s", sourceFile), destinationFile, fileutils.AssetsBox, templateData)
+	}
+	createDockerfile(t.packagingFormatName, t.dockerfileContent)
+	log.Infof("go/packaging/%s has been created. You can modify the configuration files and add it to git.", t.packagingFormatName)
+	log.Infof(fmt.Sprintf("You now can package the %s using `%s`", strings.Split(t.packagingFormatName, "-")[0], log.Au().Magenta("hover build "+t.packagingFormatName)))
+}
+
+func (t *packagingTask) Build(buildVersion string) {
+	for f := range t.dependsOn {
+		switch f.packagingFormatName {
+		case DarwinBundlePackagingTask.packagingFormatName:
+			DarwinBundlePackagingTask.Build(buildVersion)
+		case DarwinDmgPackagingTask.packagingFormatName:
+			DarwinDmgPackagingTask.Build(buildVersion)
+		case DarwinPkgPackagingTask.packagingFormatName:
+			DarwinPkgPackagingTask.Build(buildVersion)
+		case LinuxAppImagePackagingTask.packagingFormatName:
+			LinuxAppImagePackagingTask.Build(buildVersion)
+		case LinuxDebPackagingTask.packagingFormatName:
+			LinuxDebPackagingTask.Build(buildVersion)
+		case LinuxRpmPackagingTask.packagingFormatName:
+			LinuxRpmPackagingTask.Build(buildVersion)
+		case LinuxSnapPackagingTask.packagingFormatName:
+			LinuxSnapPackagingTask.Build(buildVersion)
+		case WindowsMsiPackagingTask.packagingFormatName:
+			WindowsMsiPackagingTask.Build(buildVersion)
+		}
+	}
+	projectName := pubspec.GetPubSpec().Name
+	tmpPath := getTemporaryBuildDirectory(projectName, t.packagingFormatName)
+	templateData := getTemplateData(projectName, buildVersion)
+	defer func() {
+		err := os.RemoveAll(tmpPath)
+		if err != nil {
+			log.Errorf("Could not remove temporary build directory: %v", err)
+			os.Exit(1)
+		}
+	}()
+	log.Infof("Packaging %s in %s", strings.Split(t.packagingFormatName, "-")[1], tmpPath)
+
+	if t.buildOutputDirectory != "" {
+		err := copy.Copy(build.OutputDirectoryPath(strings.Split(t.packagingFormatName, "-")[0]), executeStringTemplate(filepath.Join(tmpPath, t.buildOutputDirectory), templateData))
+		if err != nil {
+			log.Errorf("Could not copy build folder: %v", err)
+			os.Exit(1)
+		}
+	}
+	for f, l := range t.dependsOn {
```

Comment:
I have to lookup the `dependsOn` field to see that it is a `map[*packagingTask]string`, still doesn't make it clear what the value really is. `l` and the use below in filepath.Join hint that it's probably some kind of location, but it isn't very explicit.

I would suggest to rename `f, l` to `dependencyTask, targetLocation` (if that's correct?)